src/cobalt_model/pagination_config.rs

```diff
@@ -0,0 +1,155 @@
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub order: Option<SortOrder>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sort_by: Option<Vec<String>>,
+}
+
+impl PaginationConfigBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn set_include<S: Into<Option<Include>>>(self, include: S) -> Self {
+        Self {
+            include: include.into(),
+            ..self
+        }
+    }
+
+    pub fn set_per_page<S: Into<Option<i32>>>(self, per_page: S) -> Self {
+        Self {
+            per_page: per_page.into(),
+            ..self
+        }
+    }
+
+    pub fn set_permalink<S: Into<Option<String>>>(self, permalink: S) -> Self {
+        Self {
+            permalink: permalink.into(),
+            ..self
+        }
+    }
+
+    pub fn set_order<S: Into<Option<SortOrder>>>(self, order: S) -> Self {
+        Self {
+            order: order.into(),
+            ..self
+        }
+    }
+
+    pub fn set_sort_by<S: Into<Option<Vec<String>>>>(self, sort_by: S) -> Self {
+        Self {
+            sort_by: sort_by.into(),
+            ..self
+        }
+    }
+
+    pub fn merge(mut self, secondary: &PaginationConfigBuilder) -> PaginationConfigBuilder {
+        if self.include.is_none() {
+            self.include = secondary.include;
+        }
+        if self.per_page.is_none() {
+            self.per_page = secondary.per_page;
+        }
+        if self.permalink.is_none() {
+            self.permalink = secondary.permalink.clone();
+        }
+        if self.order.is_none() {
+            self.order = secondary.order;
+        }
+        if self.sort_by.is_none() {
+            self.sort_by = secondary.sort_by.clone();
+        }
+        self
+    }
+
+    pub fn build(self) -> Option<PaginationConfig> {
+        let Self {
+            include,
+            per_page,
+            permalink,
+            order,
+            sort_by,
+        } = self;
+
+        let include = include.unwrap_or(Include::None);
+        if include != Include::None {
+            let per_page = per_page.unwrap_or(DEFAULT_PER_PAGE);
+            let permalink = permalink.unwrap_or_else(|| DEFAULT_PERMALINK.to_owned());
+            let order = order.unwrap_or(SortOrder::Desc);
+            let sort_by = sort_by.unwrap_or_else(|| vec![DEFAULT_SORT.to_owned()]);
+            Some(PaginationConfig {
+                include,
+                per_page,
+                permalink,
+                order,
+                sort_by,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields, default)]
+pub struct PaginationConfig {
+    pub include: Include,
+    pub per_page: i32,
+    pub permalink: String,
+    pub order: SortOrder,
+    pub sort_by: Vec<String>,
+}
+
+impl Default for PaginationConfig {
+    fn default() -> PaginationConfig {
+        PaginationConfig {
```

Comment:
Couldn't you reduce duplication by `PaginationConfigBuilder::default().build()`?

Reply:
Removing the `impl Default` makes the `Deserialize` derivation fail:
```
--> src\cobalt_model\pagination_config.rs:133:35
    |
133 | #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
    |                                   ^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `cobalt_model::pagination_config::PaginationConfig`
    |
```

Reply:
I'm not suggesting removing it, I'm suggesting changing how its implemented.

Right now, it initializes `PaginationConfig` to match the defaulting applied in `build`.  You can reduce this duplication by

```rust
impl Default for PaginationConfig {
    fn default() -> PaginationConfig {
        PaginationConfigBuilder::default().build()
    }
}
```

Hmm, now that I look at things, maybe that won't work because `build()` will turn it into a `None`.

Which gets me thinking, why does this support serde?  I'm assuming its for doing cleaner debug dumps.  We only need `Deserialize` then and don't need `#[serde(deny_unknown_fields, default)]`

At that point, we don't need a default here which I don't think will actually serve any purpose.

Reply:
> Hmm, now that I look at things, maybe that won't work because build() will turn it into a None.
That's why I tried to removed it in the first place :)

> Which gets me thinking, why does this support serde? 

Same question for https://github.com/cobalt-org/cobalt.rs/blob/8fbca2590ec8d9a7c6490990ff5f8c937db76f88/src/cobalt_model/frontmatter.rs#L441

I think I've put `Serialize` and `Deserialize` because `Frontmatter` has them

Reply:
Mind doing a PR real quick that removes `Serialize` from anything not loaded from disk?

Reply:
Ok :)

Reply:
To get it right: `FrontmatterBuilder` *is* loaded from disk but not `Frontmatter` (as it's being created from the builder) same goes for any struct created by a builder. Am I right?

Reply:
I got fixed up, we need serialize for to_string / printing debug information.

Deserialize is what we don't need.  Only config stuff gets loaded from disk.  The rest should only ever be used programmatically.

Reply:
Sorry for the delay, I got another try this morning and I don't see how to do it: it breaks a lot of stuff.

`Frontmatter` doesn't need `Deserialize` but `FrontmatterBuilder` does and both `impl Front` which has this `for<'de> serde::Deserialize<'de>` which I don't quite understand.

Reply:
Sorry, I forgot about `Front`.  Feel free to ignore dropping `Serialize`.  I can worry about it in a future cleanup.

Reply:
Roger that, I'll go back on the PR for pagination :)

Reply:
> Hmm, now that I look at things, maybe that won't work because `build()` will turn it into a `None`.

So is there still something to do here?

Reply:
Yeah, this is fine for now.